base/asyncmap.jl

```diff
@@ -145,4 +145,27 @@ function next(itr::StreamMapIterator, state::StreamMapState)
     return (r, state)
 end
 
-iteratorsize(::Type{StreamMapIterator}) = SizeUnknown()
+iteratorsize(::Type{StreamGenerator}) = SizeUnknown()
+
+
+"""
+    asyncgenerate(f, c...) -> iterator
+
+Apply `@async f` to each element of `c`.
+
+For multiple collection arguments, apply f elementwise.
+
+Results are returned in order as they become available.
+"""
+asyncgenerate(f, c...) = StreamGenerator(f, c...)
```

Comment:
If the primary difference is whether the results are saved in place vs not, then the non in place version might not need its own separate wrapper type. It could be convenient to provide both in place and non in place API options but that would more typically be done as a lower case `!` function rather than a type constructor.

As Stefan said in the other issue, something still feels off about the proliferation of names for slight variations here.